xgi/classes/hypergraph.py

```diff
@@ -531,27 +542,53 @@ def add_edges_from(self, ebunch_to_add, **attr):
         cannot add empty edges; the method skips over them.
 
         """
-        for e in ebunch_to_add:
-            if isinstance(e[-1], dict):
-                dd = e[-1]
-                e = e[:-1]
-            else:
-                dd = {}
-            if not e:
-                continue
-
-            uid = self._edge_uid()
-            self._edge[uid] = []
-            for n in e:
+        new_edges = iter(ebunch_to_add)
+        try:
+            first_edge = next(new_edges)
+        except StopIteration:
+            return
+        try:
+            first_elem = list(first_edge)[0]
+        except TypeError:
+            first_elem = None
+
+        format1, format2, format3, format4 = False, False, False, False
+        if isinstance(first_elem, Iterable):
+            if len(first_edge) == 2 and hasattr(first_edge[1], "__hash__"):
```

Comment:
maybe easier to understand/read if checking if `dict` rather than hash?

Reply:
This is checking for the first edge, no the entire thing :)  BTW I updated to check against `Hashable` rather than `__hash__` (bc there are objects that implement __hash__ but are not hashable...)